docs/processors.md

```diff
@@ -0,0 +1,158 @@
+to a single parent, which can be either a connector or a pipeline. With that, we can say that we have the following types 
+of processors:
+1. **Source processors**: these processors only receive messages originating at a specific source connector. Source 
+processors are created by specifying the corresponding source connector as the parent entity.
+2. **Pipeline processors**: these processors receive all messages that flow through the pipeline, regardless of the
+   source or destination. Pipeline processors are created by specifying the pipeline as the parent entity.
+3. **Destination processors**: these processors receive only messages that are meant to be sent to a specific
+   destination connector. Destination processors are created by specifying the corresponding destination connector as the
+   parent entity.
+
+Given that every processor can have one (and only one) parent, processors cannot be shared. In case the same processing 
+needs to happen for different sources or destinations, you have two options:
+1. If records from all sources (or all destinations) need to be processed in the same way, then you can obviously create
+a pipeline processor
+2. If records from some, but not all, sources (or destinations) need to be processed in the same way, then you need to 
+create multiple processors (one for each source or destination) and configure them in the same way.
+
+## Adding and configuring a processor
+
+Processors are created through the `/processors` endpoint. Here's an example:
+
+```json lines
+POST /v1/processors
+{
+    // name of the processor in Conduit
+    // note that this is NOT a user-defined name for this processor
+    "name": "extractfieldpayload",
+    "parent": 
+    {
+        // type of parent: TYPE_CONNECTOR or TYPE_PIPELINE
+        "type": "TYPE_CONNECTOR",
+        // parent ID (connector ID in this case)
+        "id": "aed07589-44d8-4c68-968c-1f6c5197f13b" 
+    },
+    "config":
+    {
+        "settings":
+        {
+            // configuration map for this processor
+            "field": "name" 
+        }
+    }
+}
+```
+The request to create a processor is described in [api.swagger.json](/pkg/web/openapi/swagger-ui/api/v1/api.swagger.json).
+
+## Supported processors
+
+Conduit provides a number of built-in processors, such as filtering fields, replacing them, posting payloads to HTTP endpoints etc.
+Conduit also provides the ability to write custom processors in JavaScript.
+
+### Built-in processors
+
+An up-to-date list of all built-in processors and detailed descriptions can be found [here](https://pkg.go.dev/github.com/conduitio/conduit/pkg/processor/procbuiltin).
+An example is available in [extract-field-transform.sh](/examples/processors/extract-field-transform.sh). The script will
+set up a pipeline with the built-in extract-field processors.
+
+### JavaScript processors
+
+JavaScript processors make it possible to write custom processors. The API name for JavaScript processors (used in the
```

Comment:
I think it would be great if we mention that we use https://github.com/dop251/goja and that users can go there to find out more about what is possible and what is not.